File: 287-find-the-duplicate-number/find-the-duplicate-number.c
int findDuplicate(int* nums, int numsSize) {
    int freq[100001] = {0};
    for (int i=0; i<numsSize; i++){
        freq[nums[i]]++;
    }
    int ans = -1;
    for (int i=0; i<100001; i++){
        if (freq[i] > 1){
            ans = i;
        }
    }
    return ans;
}

File: 287-find-the-duplicate-number/find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int i = 0;

        while (i < nums.length) {
            int idx = nums[i];

            if (nums[i] == nums[idx - 1]) {
                i++;
            }

            else {
                int temp = nums[i];
                nums[i] = nums[idx - 1];
                nums[idx - 1] = temp;
            }
        }

        return nums[nums.length - 1];
    }
}